aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -652,6 +652,31 @@ module aptos_framework::fungible_asset {
         }
     }
 
+    /// Get the balance of a given store, as AggregatorSnapshot.
+    /// Allows us to obtain a balance object, without issuing a read - which would reduce parallelism.
+    ///
+    /// Note: This function will abort on FAs with `derived_balance` hook set up.
+    ///       Use `dispatchable_fungible_asset::balance` instead if you intend to work with those FAs.
+    public fun balance_snapshot<T: key>(store: Object<T>): AggregatorSnapshot<u64> acquires FungibleStore, ConcurrentFungibleBalance, DispatchFunctionStore {
```

Comment:
why we need this function?

Reply:
The existing balance function already takes a snapshot right?

Reply:
So that it can be emitted to event?

Reply:
both balance and balance_snapshot will return the same value (irrespective of whether store is concurrent or not).

but if balance is concurrent, then balance_snapshot will not create a conflict - and callers can freely store it in another resource, or emit as an event, with keeping things parallel. In the future we will allow formulas to be applied to them as well.